### ISSUE TITLE
Gregable Correctness Issues

### ISSUE THREAD

**Issue by N-Coder** [N-Coder/pqtree.js#1]
Unfortunately, we [did find](https://dl.acm.org/doi/10.1145/3611653) correctness issues with the Gregable implementation once we also tested constraints that did not only stem from planarity testing, but also arbitrary 0/1 matrices as used here. The example (Fig 16) from the paper actually also breaks [here](https://pref.tools/pqtree#01111110&00111111&11000000&00000110&00011000&01100000&00001100).

Fortunately, our correct [PC-tree implementation](https://github.com/N-Coder/pc-tree) by now also supports stand-alone compilation without the OGDF and I checked that it also compiles with emscripten.
I tried creating a fork that switched to this implementation, but unfortunately the `glue.cpp` file mentioned in `pqtree_wrapper.cpp` is missing in this repo and thus also the `_Process` function, so I couldn't build the current version of this repo myself.
If you can help me compile the current state, I can take care of switching to a correct implementation and, for the PC-trees, also correctly handling both linear and cyclic orders.

Oh, and thanks a lot for creating this awesome tool! It is really helpful and fun to play around with PQ-trees in this nice graphic and interactive way.

**Comment by DominikPeters** [comment]
Thanks for working on this! I hope to have a look on Friday, feel free to ping should I forget.

**Comment by DominikPeters** [comment]
I added glue.cpp, hope it works. This was my first time using emscripten, so I don't think anything here is best practices :)

**Comment by DominikPeters** [comment]
Yep, seemed to work for me from a fresh Github Codespace set up in this repo, running:
```bash
git clone https://github.com/emscripten-core/emsdk.git
cd emsdk
./emsdk install latest
./emsdk activate latest
source ./emsdk_env.sh
cd ../src
./build.sh
```
